src/anyp/Uri.h

```diff
@@ -193,6 +193,7 @@ bool urlIsRelative(const char *);
 char *urlMakeAbsolute(const HttpRequest *, const char *);
 char *urlRInternal(const char *host, unsigned short port, const char *dir, const char *name);
 char *urlInternal(const char *dir, const char *name);
+bool urlAppendDomain(char *host); ///< apply append_domain config to the given hostname
```

Comment:
I think it is important to document buffer size expectations here. It would not hurt to specify the return value meaning, but I do not insist on that.

```suggestion
/// Appends configured append_domain to hostname, assuming
/// the given buffer is at least SQUIDHOSTNAMELEN bytes long.
/// \returns false if and only if there is not enough space to append
bool urlAppendDomain(char *hostname);
```

Reply:
Done, with extra mention of the dotless TLD assumption.